zeroex/orderwatch/order_watcher_test.go

```diff
@@ -1027,6 +1027,243 @@ func TestOrderWatcherCleanup(t *testing.T) {
 	}
 }
 
+func TestOrderWatcherUpdateBlockHeadersStoredInDB(t *testing.T) {
+	if !serialTestsEnabled {
+		t.Skip("Serial tests (tests which cannot run in parallel) are disabled. You can enable them with the --serial flag")
+	}
+
+	teardownSubTest := setupSubTest(t)
+	defer teardownSubTest(t)
+
+	meshDB, err := meshdb.New("/tmp/leveldb_testing/" + uuid.New().String())
+	require.NoError(t, err)
+
+	ctx, cancelFn := context.WithCancel(context.Background())
+	defer cancelFn()
+	_, orderWatcher := setupOrderWatcher(ctx, t, ethRPCClient, meshDB)
+
+	// Scenario 1: Header 1 exists in DB. Get's removed and then re-added.
```

Comment:
These should all be independent test cases in order to make sure they aren't interfering with each other. If we change `updateBlockHeadersStoredInDB` to a function, it should be pretty straightforward to use table-driven tests here.